File: tests/fixtures/sample.go
package main

import (
	"fmt"
	"strings"
)

type Counter struct {
	value int
}

func NewCounter(start int) *Counter {
	return &Counter{value: start}
}

func (c *Counter) Increment() {
	c.value++
}

func greet(name string) string {
	return fmt.Sprintf("hello, %s", strings.ToLower(name))
}

func main() {
	c := NewCounter(0)
	c.Increment()
	fmt.Println(greet("World"), c.value)
}

File: tests/fixtures/sample.py
"""Sample Python file used by code-chunker tests."""

from __future__ import annotations

import os
from pathlib import Path


def greet(name: str) -> str:
    """Return a friendly greeting."""
    return f"hello, {name}"


def add(a: int, b: int) -> int:
    return a + b


class Counter:
    """A tiny stateful object."""

    def __init__(self, start: int = 0) -> None:
        self.value = start

    def increment(self) -> None:
        self.value += 1

    def decrement(self) -> None:
        self.value -= 1

    def reset(self) -> None:
        self.value = 0


CONSTANT = 42
